src/utils/tree.ts

```diff
@@ -0,0 +1,92 @@
+import { request } from '@/utils/http'
+
+
+interface ManifestIndices {
+    collectionIndex: number,
+    manifestIndex: number
+}
+
+interface ItemIndices {
+    collectionUrl: string,
+    manifestIndex: number,
+    itemIndex: number
+}
+
+export async function createTree(panels: PanelConfig[]) {
```

Comment:
As a slight improvement, this function could receive an array of collection URLs instead of dealing with panelConfig.